### PR TITLE
Clean specific bundle files

### PR DESCRIPTION
**Pull request by Mropat**
## Description

```
Usage: cg clean hk-bundle-files [OPTIONS]

  Remove files found in housekeeper bundles

Options:
  -c, --case_id TEXT
  -p, --pipeline PIPELINE
  -t, --tags TEXT          [required]
  -o, --days-old INTEGER
  -d, --dry-run            Shows cases and files that would be cleaned
  --help                   Show this message and exit.
```

### Changed

- cg clean hk-bundle-files refactored:
  - Will clean bundles older than x days
  - Will filter by pipeline as criteria
  - Will accept multiple tags as criteria
  - WIll accept case_id as criteria
  - WIll clean all bundles, not just latest bundles

### How to prepare for test
- [x] ssh to relevant server (depending on type of change)
- [x] Use stage: `us`
- [x] paxa the environment: `paxa`
- [x] install on stage (example for hasta):
`bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-cg-stage.sh [THIS-BRANCH-NAME]`

### Expected test outcome
- [ ] check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review
- [ ] code approved by
- [ ] tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a
- [ ] **MAJOR** - when you make incompatible API changes
- [x] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan
- [ ] Document in ...
- [ ] Deploy this branch
- [ ] Inform to ...

### CODE REVIEW

**Comment by henrikstranneheim** [comment]
@Mropat Nice! Looking forward to this one!

**Comment by Mropat** [comment]
```
[maryia.ropart@hasta:~] [S_main] 2m31s $ cg clean hk-bundle-files -t fastq -d -c goodmoray

Cleaning analysis goodmoray | 2021-10-07
Fetch version 2021-10-07 18:09:11 from bundle goodmoray
Version found for bundle:goodmoray; pipeline: sars-cov-2; date 2021-10-07 18:09:11
Fetching files from bundle goodmoray
Fetching files with tags in [fastq]
Fetching files from version 59350
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/923177.consensus.fa not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511130_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511130_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511131_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511131_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511132_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511132_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511133_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511133_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511134_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511134_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511135_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511135_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511136_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511136_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511137_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511137_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511138_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511138_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511139_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511139_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511140_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511140_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511141_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511141_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511142_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511142_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511143_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511143_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511144_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511144_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511145_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511145_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511146_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511146_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511147_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511147_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511148_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511148_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511149_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511149_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511150_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511150_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511151_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511151_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511152_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511152_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511153_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511153_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511154_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511154_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511155_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511155_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511156_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511156_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511157_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511157_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511158_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511158_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511159_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511159_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511160_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511160_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511161_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511161_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511162_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511162_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511163_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511163_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511165_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511165_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511166_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511166_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511169_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511169_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511170_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511170_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511171_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511171_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511172_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511172_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511173_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511173_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511174_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511174_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511175_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511175_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511176_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511176_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511177_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511177_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511178_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511178_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511179_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511179_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511180_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511180_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511181_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511181_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511182_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511182_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511225_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511225_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511184_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511184_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511185_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511185_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511186_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511186_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511187_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511187_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511188_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511188_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511189_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511189_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511190_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511190_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511191_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511191_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511192_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511192_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511193_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511193_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511194_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511194_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511196_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511196_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511197_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511197_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511198_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511198_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511199_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511199_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511200_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511200_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511201_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511201_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511202_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511202_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511203_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511203_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511204_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511204_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511205_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511205_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511206_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511206_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511207_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511207_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511208_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511208_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511209_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511209_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511210_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511210_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511211_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511211_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511212_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511212_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511213_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511213_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511214_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511214_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511215_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511215_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511216_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511216_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511217_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511217_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511218_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511218_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511219_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511219_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511220_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511220_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511221_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511221_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511222_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511222_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511223_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511223_2.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511224_1.fastq.gz not on disk.
File /home/proj/stage/housekeeper-bundles/goodmoray/2021-10-07/01_SE100_21CS511224_2.fastq.gz not on disk.
Process freed 0.0GB. Dry run: True

```

**Review by henningonsbring — APPROVED** [review]

File: cg/store/api/status.py [hunk 1]
```diff
@@ -589,6 +589,23 @@ def analyses_to_clean(
 
         return records
 
+    def analyses_before_date(
+        self,
+        case_id: Optional[str] = None,
+        before: Optional[datetime] = datetime.now(),
+        pipeline: Optional[Pipeline] = None,
+    ) -> Query:
+        """Fetch all analyses older than certain date."""
```
Comment (henrikstranneheim) [inline cg/store/api/status.py:598]:
```suggestion
        """Fetch all analyses older than certain date"""
```

File: cg/cli/clean.py [hunk 2]
```diff
@@ -106,46 +108,68 @@ def scout_finished_cases(
...
                 continue
-            LOG.info("Will remove %s", file_path)
-            if (yes or click.confirm("Do you want to remove this file?")) and not dry_run:
-                hk_file.delete()
-                housekeeper_api.commit()
-                if file_path.exists():
-                    file_path.unlink()
-                LOG.info("File removed")
+            LOG.info(f"File {file_path} found on disk.")
+            file_size = file_path.stat().st_size
```
Comment (henrikstranneheim) [inline cg/cli/clean.py:165]: Typehint

File: cg/cli/clean.py [hunk 3]
```diff
@@ -106,46 +108,68 @@ def scout_finished_cases(
...
-        last_version: Optional[hk_models.Version] = housekeeper_api.last_version(bundle=case_id)
-        if not last_version:
-            continue
-        last_version_file_paths: List[Path] = [
-            Path(hk_file.full_path)
-            for hk_file in housekeeper_api.get_files(
-                bundle=case_id, tags=None, version=last_version.id
+    date_threshold: datetime = datetime.now() - timedelta(days=days_old)
+
+    analyses = status_db.analyses_before_date(
```
Comment (henrikstranneheim) [inline cg/cli/clean.py:133]: typehint

File: cg/cli/clean.py [hunk 4]
```diff
@@ -106,46 +108,68 @@ def scout_finished_cases(
...
+        case_id=case_id, before=date_threshold, pipeline=pipeline
+    )
+    size_cleaned = 0
```
Comment (henrikstranneheim) [inline cg/cli/clean.py:136]: typehint

File: cg/cli/clean.py [hunk 5]
```diff
@@ -106,46 +108,68 @@ def scout_finished_cases(
...
+    for analysis in analyses:
+        LOG.info(f"Cleaning analysis {analysis}")
+        bundle_name = analysis.family.internal_id
```
Comment (henrikstranneheim) [inline cg/cli/clean.py:139]: Would be great with some more typehints throughout

File: cg/store/api/status.py [hunk 6]
```diff
@@ -589,6 +589,23 @@ def analyses_to_clean(
 
         return records
 
+    def analyses_before_date(
```
Comment (henrikstranneheim) [inline cg/store/api/status.py:592]:
```suggestion
    def get_analyses_before_date(
```

**Review by henrikstranneheim — APPROVED** [review]
👍 ! Excellent

File: cg/store/api/status.py (hunk 6 above)
Comment (Mropat) [reply cg/store/api/status.py:592]: fixed

File: cg/cli/clean.py (hunk 5 above)
Comment (Mropat) [reply cg/cli/clean.py:139]: done

**Pull request merged**